### PR TITLE
fix: missing variable declarations for masters-only scenario

### PR DESCRIPTION
**Pull request by jackfrancis**
**Reason for Change**:

Master-only scenario has missing ARM variable definition in template generation refactor execution flow.

This is the equivalent in the current template generation implementation:

https://github.com/Azure/aks-engine/blob/master/parts/k8s/kubernetesmastervars.t#L229

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
@tariq1890 FYI

**Comment by jackfrancis** [comment]
Manual tests suggest that this change is all that is needed to fix "masters only" cluster configuration scenarios using the refactored template generation flow.

File: pkg/engine/armvariables.go [hunk 1]
```diff
@@ -316,6 +316,7 @@ func getK8sMasterVars(cs *api.ContainerService) map[string]interface{} {
 		if hasAgentPool {
 			masterVars["primaryAvailabilitySetName"] = fmt.Sprintf("[concat('%s-availabilitySet-',parameters('nameSuffix'))]", profiles[0].Name)
 		}
+		masterVars["primaryAvailabilitySetName"] = ""
```
Comment (tariq1890) [inline pkg/engine/armvariables.go]: Definitely looks like I have missed this. Thanks :)

**Comment by serbrech** [comment]
lgtm

**Pull request merged**